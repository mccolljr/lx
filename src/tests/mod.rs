use std::process::Command;

#[test]
fn test_snapshots() {
    glob!("testdata/*.lx", |path| {
        println!("testing {:?}", path);
        let output = Command::new("cargo")
            .arg("run")
            .arg(path)
            .output()
            .expect("failed to execute test");
        let output_str =
            String::from_utf8(output.stdout).expect("oops, output not utf8");
        assert_snapshot!(output_str);
    });
}
