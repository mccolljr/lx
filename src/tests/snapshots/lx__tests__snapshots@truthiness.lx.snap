---
source: src/tests/mod.rs
expression: output_str
input_file: src/tests/testdata/truthiness.lx
---
null is truthy?        false
true is truthy?        true
false is truthy?       false
1 is truthy?           true
0 is truthy?           false
1.0 is truthy?         true
0.0 is truthy?         false
"a" is truthy?         true
"" is truthy?          false
[1] is truthy?         true
[] is truthy?          false
{a: 1} is truthy?      true
{} is truthy?          false
(fn(){}) is truthy?    true
(print) is truthy?     true
empty iter is truthy?  true
iter is truthy?        true
OK!

